### PR TITLE
Initial version of v1beta3 KubeOneCluster

### PR DESCRIPTION
**Pull request by kron4eg**
**What this PR does / why we need it**:
This is a base code-infrastructure to further improve and apply out plans for v1beta3 API. It's identical to v1beta2 except docker container runtime.

**What type of PR is this?**
/kind feature
/kind chore

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
Initial version of v1beta3 KubeOneCluster
```

**Documentation**:

```documentation
NONE
```

### CODE REVIEW

File: pkg/apis/kubeone/v1beta3/register.go [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
-Copyright 2020 The KubeOne Authors.
+Copyright 2021 The KubeOne Authors.
```
Comment (xmudrii) [inline pkg/apis/kubeone/v1beta3/register.go:2]: We should change license headers to 2024 for v1beta3 (affects other files in this package too).

File: pkg/apis/kubeone/config/migrate_test.go
Comment (xmudrii) [inline pkg/apis/kubeone/config/migrate_test.go:1]: Why are tests for the migrate command removed? I think those would still be useful.

File: pkg/cmd/config-images.go [hunk 2]
```diff
@@ -120,23 +120,29 @@ func listImages(opts *listImagesOpts) error {
...
-			return err
-		}
-
-		overRegGetter := images.WithOverwriteRegistryGetter(func() string {
-			if rc := conf.RegistryConfiguration; rc != nil {
-				return rc.OverwriteRegistry
+		func() {
+			var conf kubeonev1beta3.KubeOneCluster
+			if err := yaml.Unmarshal(configBuf, &conf); err != nil {
+				// we don't care if there was an error, just proceed like there was no manifest file given at all
```
Comment (xmudrii) [inline pkg/cmd/config-images.go:126]: Why is this change needed?

Comment (kron4eg) [reply pkg/cmd/config-images.go]: Initial idea was that "since we don't care if file exists, maybe we don't care about the unmarshal error ether". But yeah, those are different cases. Returned back as it was.

File: pkg/apis/kubeone/v1beta3/register.go (hunk 1 above)
Comment (kron4eg) [reply pkg/apis/kubeone/v1beta3/register.go]: Fixed

**Review by xmudrii — APPROVED** [review]
/lgtm
/approve
